src/gps/gps.c

```diff
@@ -177,6 +201,7 @@ void GPS_sample_update(GpsSample *newSample)
     updateFullDateTime(newSample);
     g_gpsSnapshot.deltaFirstFix = newSample->time - g_timeFirstFix;
     g_gpsSnapshot.previousPoint = prevPoint;
+    g_gpsSnapshot.bearing = gps_bearing(&prevPoint, &(newSample->point));
```

Comment:
I agree that it wouldn't be appropriate in this case. Neither would speed based, as the demo data showed glitches on reasonably high speeds (> 10 mph) I think the only reliable way to do it is to only calculate it based on a distance threshold.